[EDR Workflows] Elastic defend policy migration for advanced properties

szwarckonrad opened the pull request:
Adds a migration for advanced package policy fields, based on values provided here -  https://github.com/elastic/security-team/issues/9650#issuecomment-2176499768

Combines `x-pack/plugins/security_solution/public/management/pages/policy/models/advanced_policy_schema.ts` changes from:

https://github.com/elastic/kibana/pull/184725
https://github.com/elastic/kibana/pull/184786

szwarckonrad commented:
/ci

szwarckonrad commented:
/ci

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

dasansol92 reviewed (changes requested):
Blocking this pr to ensure it doesn't gets accidentally merged cc: @szwarckonrad

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

SiddharthMantri reviewed (approved):
`PACKAGE_POLICY_SAVED_OBJECT_TYPE` is not an encrypted SO type. No changes for Platform security to verify.

ferullo reviewed (commented):
`advanced_policy_schema.ts` LGTM

kpollich reviewed (approved):
Fleet migration LGTM

gergoabraham reviewed (commented):
looks great! tried to proofread it (since there's no type safety here), didn't find any issue. and thanks for the detailed tests! 🚀 

one question though: some weeks ago I added a new folder to let our mindsets move from migrations to model versions, so instead Kibana version related migrations like `v8.15`, we can add backfills related to model versions to this folder:
```
x-pack/plugins/fleet/server/saved_objects/model_versions/security_solution/
```

what do you think about moving the new backfill function in this folder?

my thoughts on moving it:
- contra: this is actually related to Kibana version instead of model version
- pro: still I think it's better to have all new backfills in one place, and we still can indicate the motivation in the name of the backfill function and filename (e.g. `v13_endpoint_integration_backfills_for_agent_version_8_15` or something better)

szwarckonrad commented:
> looks great! tried to proofread it (since there's no type safety here), didn't find any issue. and thanks for the detailed tests! 🚀
> 
> one question though: some weeks ago I added a new folder to let our mindsets move from migrations to model versions, so instead Kibana version related migrations like `v8.15`, we can add backfills related to model versions to this folder:
> 
> ```
> x-pack/plugins/fleet/server/saved_objects/model_versions/security_solution/
> ```
> 
> what do you think about moving the new backfill function in this folder?
> 
> my thoughts on moving it:
> 
> * contra: this is actually related to Kibana version instead of model version
> * pro: still I think it's better to have all new backfills in one place, and we still can indicate the motivation in the name of the backfill function and filename (e.g. `v13_endpoint_integration_backfills_for_agent_version_8_15` or something better)

Makes sense! Moved. c79f11bdd7b9d2f4bfd58b4671e649abb7b21cf4

gergoabraham reviewed (approved):
🚀

szwarckonrad commented:
Tested migration locally:
1. Checked out `main`, ran ES + Kibana
2. Stopped Kibana
3. Checked out this branch, started Kibana

Everything looks as expected, existing policies were updated. Checked through API and Kibana's UI Agent Policy details.

On `main`
![Screenshot 2024-06-28 at 13 14 32](https://github.com/elastic/kibana/assets/29123534/d4f5118c-5fd8-40c3-ba06-e7800cae5d9b)

On this branch 
![Screenshot 2024-06-28 at 13 15 34](https://github.com/elastic/kibana/assets/29123534/82b8983d-238a-4cd8-b512-2c649ce4e3d7)

dasansol92 reviewed (approved):
Approving to unblock this pr, not tested. Thanks @szwarckonrad !
